views/js/test/runner/plugins/tools/answerElimination/plugin/test.js

```diff
@@ -0,0 +1,309 @@
+                                    })
+                                    .catch(function(err) {
+                                        console.log(err);
+                                        assert.ok(false, 'The destroy method must not fail');
+                                        QUnit.start();
+                                    });
+                            })
+                            .catch(function(err) {
+                                console.log(err);
+                                assert.ok(false, 'The enable method must not fail');
+                                QUnit.start();
+                            });
+                    })
+                    .catch(function(err) {
+                        console.log(err);
+                        assert.ok(false, 'The render method must not fail');
+                        QUnit.start();
+                    });
+            })
+            .catch(function(err) {
+                console.log(err);
+                assert.ok(false, 'The init method must not fail');
+                QUnit.start();
+            });
+    });
+
+
+    QUnit.asyncTest('eliminator.enable', function(assert) {
+        var runner = runnerFactory(providerName);
+        var _eliminator = eliminator(runner, runner.getAreaBroker());
+
+        _eliminator.init()
+            .then(function() {
+                assert.equal(_eliminator.getState('init'), true, 'Eliminator is initialised');
+                assert.equal(_eliminator.getState('enabled'), false, 'Eliminator is disabled');
+
+                _eliminator.enable()
+                    .then(function() {
+                        assert.equal(_eliminator.getState('enabled'), true, 'Eliminator is enabled');
+
+                        _eliminator.destroy()
+                            .then(function() {
+                                assert.equal(_eliminator.getState('init'), false, 'Eliminator is destroyed');
+
+                                QUnit.start();
+                            })
+                            .catch(function(err) {
+                                console.log(err);
+                                assert.ok(false, 'The destroy method must not fail');
+                                QUnit.start();
+                            });
+                    })
+                    .catch(function(err) {
+                        console.log(err);
+                        assert.ok(false, 'The enable method must not fail');
+                        QUnit.start();
+                    });
+            })
+            .catch(function(err) {
+                console.log(err);
+                assert.ok(false, 'The init method must not fail');
+                QUnit.start();
+            });
+    });
+
+
+    QUnit.asyncTest('eliminator.disable', function(assert) {
+        var runner = runnerFactory(providerName);
+        var _eliminator = eliminator(runner, runner.getAreaBroker());
+
+        _eliminator.init()
```

Comment:
This is just a tip, you don't really need to nest the Promises and to check the `catch` at each level:

```js
 _eliminator
  .init()
  .then(function() {
       return  _eliminator.enable();
  })
  .then(function(){
      return  _eliminator.disable();
  })
  .catch(function(err){

  });
   
```